### DISCUSSION TITLE
Is it possible to make renovate respect potential .NET Version compatibility issues?

### DISCUSSION THREAD

**Discussion by SteveW94** [renovatebot/renovate#14387] [Request Help] [answered]
Hello,

we update our big ASP.NET/Core Solution with renovate which mostly works great.

Anyhow we have to keep great attention on some Dependencys cause we were not able to migrate on the latest .NET Version yet.
Most of those are in the .NET Monorepo PR but some are also scattered around in some other PRs (especially some Microsoft.Extensions packages)

So when you accidentially merge one of those, you may get weird errors due to incompatibility.

Nontheless the required .NET version is mostly documented directly on https://www.nuget.org/

Is there some possibility to filter packages according to this or at least display some hint in the PR when the current .NET Version may be incompatible?
This could prevent a bit of too fast merging even when the CI branch builds green, what it oddly does often in those cases...

Thanks :)

**Comment by rarkins** [comment, marked as answer]
This functionality would need to be added. It needs a few things:
- we can detect your project's dot net constraints 
- we can detect each dependency version's dot net constraints 
- we correlate them and exclude mismatches

I'd say that it's also important that we can *update* your project's dot net constraints so that you don't eventually remain stuck on an old version with old dependencies yet with zero PRs from Renovate.

If you're willing to research the above then we could definitely accept a feature request due this

**Reply by FraserGlasgow** [reply]
do you know if any work has been done on this, since this discussion?

**Reply by viceice** [reply]
no

**Reply by omlnaut** [reply]
I've been trying to find information on this. Can anyone tell me if this is possible now?

**Reply by FraserGlasgow** [reply]
i dont think so. We have added a bunch of custom exclusions to get round it, but its obviously a overhead to keep them up to date

**Reply by davidwallis** [reply]
@FraserGlasgow  can you share any examples of how you did this? (just interested in starting point)

**Reply by Piedone** [reply]
Perhaps useful, related: https://github.com/renovatebot/renovate/discussions/33104
